compiler/src/jdk.graal.compiler.test/src/jdk/graal/compiler/hotspot/test/DeferredBarrierAdditionTest.java

```diff
@@ -98,7 +99,7 @@ protected void testHelper(final String snippetName, final int expectedBarriers,
             new GuardLoweringPhase().apply(graph, midContext);
             new MidTierLoweringPhase(createCanonicalizerPhase()).apply(graph, midContext);
             graph.getGraphState().setAfterFSA();
-            new WriteBarrierAdditionPhase().apply(graph, midContext);
+            new WriteBarrierAdditionPhase(GraphState.StageFlag.MID_TIER_BARRIER_ADDITION).apply(graph, midContext);
```

Comment:
Maybe it's better for the default constructor to just assume `MID_TIER_BARRIER_ADDITION`?  Part of me wants to drop the `MID_TIER_` from that name as well since eventually we'll be back to only having one `WriteBarrierAdditionPhase`.